StripeConnect/StripeConnect/Source/Internal/Webview/MessageSenders/MessageSendable.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  MessageSendable.swift
+//  StripeConnect
+//
+//  Created by Chris Mays on 8/2/24.
+//
+
+import Foundation
+
+protocol MessageSender {
+    associatedtype MessageType: Codable & Equatable
+    var name: String { get }
+    var message: MessageType { get }
+}
+
+extension MessageSender {
+    func toJavascriptMessage() -> String? {
+        let encoder = JSONEncoder()
+        // Ensure keys are sorted for test stability.
+        encoder.outputFormatting = .sortedKeys
+        guard let jsonData = try? encoder.encode(message),
+              let jsonString = String(data: jsonData, encoding: .utf8) else {
+            return nil
```

Comment:
optional: thoughts about throwing an error instead of returning nil? Downside is then we have to throw a custom error if `String(data:)` returns nil

Reply:
If we think handling the issue at a higher level will help we can. Otherwise I can just add a todo to log this to our analytics as a potential failure.

Reply:
I don't think it matters too much if we handle it at a higher level vs. this one, but if we end up having to handle other errors at the higher level then it'll save us from having to do a `guard-let`.

I think it's reasonable to punt on it for now and we can always update it later if it makes things easier.